collectors/cveorg/collectors.py

```diff
@@ -0,0 +1,328 @@
+
+    # From https://cveproject.github.io/cve-schema/schema/docs/#oneOf_i0_containers_cna_descriptions_contains_lang
+    EN_LANG = r"^en([_-][A-Za-z]{4})?([_-]([A-Za-z]{2}|[0-9]{3}))?$"
+
+    def __init__(self) -> None:
+        """
+        Initiate collector.
+        """
+        super().__init__()
+
+        if self.snippet_creation_enabled is None:
+            self.snippet_creation_enabled = SNIPPET_CREATION_ENABLED
+
+    def clone_repo(self) -> None:
+        """
+        Clone the cvelistV5 repository if it does not exist.
+        """
+        if os.path.isdir(self.REPO_PATH):
+            return
+
+        logger.info("Cloning the cvelistV5 repository...")
+
+        cmd_parts = ("/usr/bin/git", "clone", self.REPO_URL, self.REPO_PATH)
+        # Set shell=False to run the command directly, so that Bash injection isn't possible.
+        result = Cmd.run(
+            cmd_parts, cwd=CELERY_PVC_PATH, fail_silently=True, shell=False
+        )
+        logger.debug(result.stdout)
+
+        if result.returncode == 0:
+            logger.info("The repository was successfully cloned.")
+        else:
+            # We fail silently above, because "git clone" often has stderr output
+            # This doesn't indicate a real failure, but Cmd.run() will treat it like one
+            # So we disable this behavior, then check only the return code here instead
+            msg = f"Failed to clone the repository. Returned code: {result.returncode}."
+            logger.error(msg)
+            raise CVEorgCollectorException(msg)
+
+    def update_repo(self) -> None:
+        """
+        Update the already existing cvelistV5 repository.
+        """
+        logger.info("Updating the cvelistV5 repository...")
+
+        cmd_parts = (
+            "/usr/bin/git",
+            # Our namespaced user is trying to update files on a PVC
+            # which we have write access to, but git complains the owner doesn't match
+            "-c",
+            f"safe.directory={self.REPO_PATH}",
+            "pull",
+        )
+        # Set shell=False to run the command directly, so that Bash injection isn't possible.
+        result = Cmd.run(cmd_parts, cwd=self.REPO_PATH, fail_silently=True, shell=False)
+        logger.debug(result.stdout)
+
+        if result.returncode == 0:
+            logger.info("The repository was successfully updated.")
+        else:
+            # We fail silently above, because "git pull" often has stderr output
+            # This doesn't indicate a real failure, but Cmd.run() will treat it like one
+            # So we disable this behavior, then check only the return code here instead
+            msg = (
+                f"Failed to update the repository. Returned code: {result.returncode}."
+            )
+            logger.error(msg)
+            raise CVEorgCollectorException(msg)
+
+    def get_repo_changes(self) -> (str, timezone.datetime):
```

Comment:
I would probably make tuple type hinting consistent, here you are using (x,x) and couple of lines below you are using tuple[x,x]. Both are valid, just keep it consistent.